.github/workflows/ci.yml

```diff
@@ -342,3 +341,54 @@ jobs:
           payload: |
             {"text":"Enterprise test failures on ${{ github.ref_name }}","blocks":[{"type":"header","text":{"type":"plain_text","text":":rotating_light: Enterprise test failures :rotating_light:","emoji":true}},{"type":"divider"},{"type":"section","text":{"type":"mrkdwn","text":"test(s) failed on ${{ github.ref_name }}"},"accessory":{"type":"button","text":{"type":"plain_text","text":"View Failing Workflow","emoji":true},"url":"${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}"}}]}
     
+  test-summary:
+    name: Go test failures
+    runs-on: ubuntu-latest
+    if: success() || failure() || needs.tests-completed.result == 'skipped'
+    needs:
+      - test-go
+      - test-go-fips
+      - test-go-race
+      - tests-completed
```

Comment:
I'm curious, when does test-completed get skipped?

Reply:
`test-completed` gets skipped each time all of the Go tests are successful. It's not really an issue, since GitHub treats `skipped` as successful. 

It happens due to the changes we've recently introduced to the `if` clause in `tests-completed`:
https://github.com/hashicorp/vault/blob/16c52f3dfeff54e848cfb2485b10bff29abd1411/.github/workflows/ci.yml#L283-L286
The conditional above returns `false` when all of the tests are successful. I assumed that that logic was intentional, since it means that auto-merges behave exactly how we want them.

[See an example here (note, it's a CI run from a different PR that I just had handy).](https://github.com/hashicorp/vault/actions/runs/5486041968/jobs/9995873076)

Reply:
oof, that was not intentional 😅 . good to know, I need to fix that. Looks like we skip fips tests because of that as well, which is also a problem